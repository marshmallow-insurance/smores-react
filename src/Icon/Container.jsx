import React from 'react'
import { Icon } from './Icon'
import styled from 'styled-components'

export const Container = () => {
  return (
    <div>
      <Wrapper>
        <span>address-mta</span>
        <Icon render="address-mta" />
      </Wrapper>
      <Wrapper>
        <span>alert-engine</span>
        <Icon render="alert-engine" />
      </Wrapper>
      <Wrapper>
        <span>arrow</span>
        <Icon render="arrow" />
      </Wrapper>
      <Wrapper>
        <span>arrow-curved</span>
        <Icon render="arrow-curved" />
      </Wrapper>
      <Wrapper>
        <span>at</span>
        <Icon render="at" />
      </Wrapper>
      <Wrapper>
        <span>breakdown</span>
        <Icon render="breakdown" />
      </Wrapper>
      <Wrapper>
        <span>briefcase</span>
        <Icon render="briefcase" />
      </Wrapper>
      <Wrapper>
        <span>burger</span>
        <Icon render="burger" />
      </Wrapper>
      <Wrapper>
        <span>calendar</span>
        <Icon render="calendar" />
      </Wrapper>
      <Wrapper>
        <span>camera</span>
        <Icon render="camera" />
      </Wrapper>
      <Wrapper>
        <span>car</span>
        <Icon render="car" />
      </Wrapper>
      <Wrapper>
        <span>car-claim</span>
        <Icon render="car-claim" />
      </Wrapper>
      <Wrapper>
        <span>car-mta</span>
        <Icon render="car-mta" />
      </Wrapper>
      <Wrapper>
        <span>card</span>
        <Icon render="card" />
      </Wrapper>
      <Wrapper>
        <span>caret</span>
        <Icon render="caret" />
      </Wrapper>
      <Wrapper>
        <span>charge</span>
        <Icon render="charge" />
      </Wrapper>
      <Wrapper>
        <span>chat</span>
        <Icon render="chat" />
      </Wrapper>
      <Wrapper>
        <span>chat-bubble</span>
        <Icon render="chat-bubble" />
      </Wrapper>
      <Wrapper>
        <span>city</span>
        <Icon render="city" />
      </Wrapper>
      <Wrapper>
        <span>clock</span>
        <Icon render="clock" />
      </Wrapper>
      <Wrapper>
        <span>contact-email</span>
        <Icon render="contact-email" />
      </Wrapper>
      <Wrapper>
        <span>cross</span>
        <Icon render="cross" />
      </Wrapper>
      <Wrapper>
        <span>doc</span>
        <Icon render="doc" />
      </Wrapper>
      <Wrapper>
        <span>doc-download</span>
        <Icon render="doc-download" />
      </Wrapper>
      <Wrapper>
        <span>download</span>
        <Icon render="download" />
      </Wrapper>
      <Wrapper>
        <span>edit</span>
        <Icon render="edit" />
      </Wrapper>
      <Wrapper>
        <span>europe</span>
        <Icon render="europe" />
      </Wrapper>
      <Wrapper>
        <span>file</span>
        <Icon render="file" />
      </Wrapper>
      <Wrapper>
        <span>fire</span>
        <Icon render="fire" />
      </Wrapper>
      <Wrapper>
        <span>house</span>
        <Icon render="house" />
      </Wrapper>
      <Wrapper>
        <span>id</span>
        <Icon render="id" />
      </Wrapper>
      <Wrapper>
        <span>info</span>
        <Icon render="info" />
      </Wrapper>
      <Wrapper>
        <span>injury</span>
        <Icon render="injury" />
      </Wrapper>
      <Wrapper>
        <span>intercom</span>
        <Icon render="intercom" />
      </Wrapper>
      <Wrapper>
        <span>key</span>
        <Icon render="key" />
      </Wrapper>
      <Wrapper>
        <span>minus</span>
        <Icon render="minus" />
      </Wrapper>
      <Wrapper>
        <span>money-bag</span>
        <Icon render="money-bag" />
      </Wrapper>
      <Wrapper>
        <span>new-window</span>
        <Icon render="new-window" />
      </Wrapper>
      <Wrapper>
        <span>padlock</span>
        <Icon render="padlock" />
      </Wrapper>
      <Wrapper>
        <span>people</span>
        <Icon render="people" />
      </Wrapper>
      <Wrapper>
        <span>person</span>
        <Icon render="person" />
      </Wrapper>
      <Wrapper>
        <span>plus</span>
        <Icon render="plus" />
      </Wrapper>
      <Wrapper>
        <span>plus-circle</span>
        <Icon render="plus-circle" />
      </Wrapper>
      <Wrapper>
        <span>policy</span>
        <Icon render="policy" />
      </Wrapper>
      <Wrapper>
        <span>pound-filled-circle</span>
        <Icon render="pound-filled-circle" />
      </Wrapper>
      <Wrapper>
        <span>pound-outline</span>
        <Icon render="pound-outline" />
      </Wrapper>
      <Wrapper>
        <span>profile</span>
        <Icon render="profile" />
      </Wrapper>
      <Wrapper>
        <span>question</span>
        <Icon render="question" />
      </Wrapper>
      <Wrapper>
        <span>refund</span>
        <Icon render="refund" />
      </Wrapper>
      <Wrapper>
        <span>renewal</span>
        <Icon render="renewal" />
      </Wrapper>
      <Wrapper>
        <span>schedule-quote</span>
        <Icon render="schedule-quote" />
      </Wrapper>
      <Wrapper>
        <span>search</span>
        <Icon render="search" />
      </Wrapper>
      <Wrapper>
        <span>send-message</span>
        <Icon render="send-message" />
      </Wrapper>
      <Wrapper>
        <span>shield</span>
        <Icon render="shield" />
      </Wrapper>
      <Wrapper>
        <span>tick</span>
        <Icon render="tick" />
      </Wrapper>
      <Wrapper>
        <span>tow-truck</span>
        <Icon render="tow-truck" />
      </Wrapper>
      <Wrapper>
        <span>trash</span>
        <Icon render="trash" />
      </Wrapper>
      <Wrapper>
        <span>up-down</span>
        <Icon render="up-down" />
      </Wrapper>
      <Wrapper>
        <span>upload-email</span>
        <Icon render="upload-email" />
      </Wrapper>
      <Wrapper>
        <span>village</span>
        <Icon render="village" />
      </Wrapper>
      <Wrapper>
        <span>warning</span>
        <Icon render="warning" />
      </Wrapper>
      <Wrapper>
        <span>wheel-flat</span>
        <Icon render="wheel-flat" />
      </Wrapper>
      <Wrapper>
        <span>windscreen</span>
        <Icon render="windscreen" />
      </Wrapper>
      <Wrapper>
        <span>windscreen-repair</span>
        <Icon render="windscreen-repair" />
      </Wrapper>
      <Wrapper>
        <span>windscreen-replacement</span>
        <Icon render="windscreen-replacement" />
      </Wrapper>
    </div>
  )
}

const Wrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;
  width: 250px;
`
