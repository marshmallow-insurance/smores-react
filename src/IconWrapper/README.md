# < IconWrapper />

## Props:

```ts
className?: string
children: ReactNode
render: string /* icon set */
size?: number /* icon size */
t?: string /* top position */
r?: string /* right position */
b?: string /* bottom position */
l?: string /* left position */
```

## Usage

This component is designed to be used as wrapper that will add the rendered icon
as a position absolute element on top of the children.
You can specify the position you want from the props and it will default to top/left.

```js
import {IconWrapper} from '@mrshmllw/smores-react';

const App = () => (
  <IconWrapper {...props}>
    {children}
  </IconWrapper>
);
```

## Icon set:

* included
* excluded
