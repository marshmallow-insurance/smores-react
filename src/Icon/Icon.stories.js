import React from 'react';
import {storiesOf} from '@storybook/react';

import {text} from '@storybook/addon-knobs';

import {Icon} from './Icon';

storiesOf('Icon', module).addWithJSX('default', () => (
  <Icon
    render={text('Render', 'contact-at')}
    size={text('Size (in px)', '32')}
    color={text('Color', 'blue7')}
  />
));
