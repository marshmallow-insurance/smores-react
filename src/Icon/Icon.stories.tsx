import React from 'react'
import { Icon, IconProps } from './Icon'
import { Container } from './Container'

export default {
  title: 'Icon',
  component: Icon,
}

const Template = (props: IconProps) => <Icon {...props} />

export const Default = Template.bind({})

Default.args = {
  render: 'info',
  size: 32,
  color: 'liquorice',
  rotate: '0',
  mt: '0',
  mr: '0',
  mb: '0',
  ml: '0',
}

const CollectionTemplate = () => <Container />

export const Collection = CollectionTemplate.bind({})

Collection.args = {
  size: 24,
  color: 'liquorice',
  rotate: '0',
  mt: '0',
  mr: '0',
  mb: '',
  ml: '0',
}
