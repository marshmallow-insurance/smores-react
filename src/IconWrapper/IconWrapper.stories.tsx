import React from 'react'
import { IconWrapper, IconWrapperProps } from './IconWrapper'
import { Icon } from '../Icon'

export default {
  title: 'IconWrapper',
  component: IconWrapper,
}

const Template = (props: IconWrapperProps) => (
  <IconWrapper {...props}>
    <Icon render="wheel-flat" />
  </IconWrapper>
)

export const Default = Template.bind({})

Default.args = {
  render: 'included',
  size: 16,
  t: '0',
  r: '0',
  b: '0',
  l: '0',
}

export const Excluded = Template.bind({})

Excluded.args = {
  render: 'excluded',
  l: '0',
  t: '0',
}

export const BottomRight = Template.bind({})

BottomRight.args = {
  r: '0',
  b: '0',
}

export const BottomLeft = Template.bind({})

BottomLeft.args = {
  l: '0',
  b: '0',
}

export const TopRight = Template.bind({})

TopRight.args = {
  r: '0',
  t: '0',
}

export const TopLeft = Template.bind({})

TopLeft.args = {
  l: '0',
  t: '0',
}
