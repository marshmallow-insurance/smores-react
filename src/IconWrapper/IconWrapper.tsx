import React, { FC, ReactNode } from 'react'
import styled, { css } from 'styled-components'

import { Box } from '../Box'
import { Icon } from '../Icon'
import { MarginProps } from '../utils/space'

export type IconWrapperProps = {
  /** className attribute to apply classes from props */
  className?: string
  /* */
  children: ReactNode
  /** specify what Icon to render  */
  render: string
  /** set size of the Icon  */
  size?: number
  /** TOP-RIGHT-BOTTOM-LEFT position of the Icon */
  t?: string
  r?: string
  b?: string
  l?: string
} & MarginProps

export const IconWrapper: FC<IconWrapperProps> = ({
  className = '',
  children,
  size = 16,
  render = 'included',
  t = 'auto',
  r = 'auto',
  b = 'auto',
  l = 'auto',
  ...marginProps
}) => (
  <Container className={className} {...marginProps}>
    {children}
    <IconContainer size={size} t={t} r={r} b={b} l={l}>
      {render === 'included' && (
        <Icon render="included" size={size} color="apple" />
      )}
      {render === 'excluded' && (
        <Icon render="excluded" size={size} color="strawberry" />
      )}
    </IconContainer>
  </Container>
)

const Container = styled(Box)`
  position: relative;
  width: fit-content;
`

interface IIcon {
  size: number
  t: string
  r: string
  b: string
  l: string
}

const IconContainer = styled.div<IIcon>(
  ({ size, t, r, b, l }) => css`
    position: absolute;
    width: 100%;
    max-width: ${size}px;
    height: ${size}px;
    top: ${t};
    right: ${r};
    bottom: ${b};
    left: ${l};
  `,
)
