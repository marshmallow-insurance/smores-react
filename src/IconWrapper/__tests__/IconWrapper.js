import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import { IconWrapper } from '../IconWrapper'
import { Icon } from '../../Icon'

test('renders', () => {
  const props = {
    render: 'included',
    size: 16,
    t: 'auto',
    r: 'auto',
    b: 'auto',
    l: 'auto',
  }
  const { container } = render(
    <IconWrapper {...props}>
      <Icon render="wheel-flat" />
    </IconWrapper>,
  )
  expect(container.firstChild).toMatchSnapshot()
})
