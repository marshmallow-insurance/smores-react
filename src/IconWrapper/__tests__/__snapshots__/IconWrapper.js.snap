// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c0 {
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c2 {
  position: absolute;
  width: 100%;
  max-width: 16px;
  height: 16px;
  top: auto;
  right: auto;
  bottom: auto;
  left: auto;
}

<div
  class="c0"
>
  <span
    class="c1"
    color="liquorice"
    rotate="0"
    size="24"
  >
    <svg
      fill="none"
      height="100%"
      viewBox="0 0 24 24"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 12a1 1 0 100 2 1 1 0 000-2zm-3 1a3 3 0 116 0 3 3 0 01-6 0z"
        fill="#112035"
        fill-rule="evenodd"
      />
      <path
        clip-rule="evenodd"
        d="M12.95 13.309a1 1 0 011.261-.642l4.755 1.545a1 1 0 11-.618 1.902l-4.755-1.545a1 1 0 01-.642-1.26zM16.702 6.528a1 1 0 01.221 1.397l-2.938 4.045a1 1 0 01-1.618-1.176l2.938-4.045a1 1 0 011.397-.221zM7.298 6.528a1 1 0 011.397.221l2.938 4.045a1 1 0 01-1.618 1.176L7.077 7.925a1 1 0 01.221-1.397zM11.05 13.309a1 1 0 01-.643 1.26l-4.755 1.545a1 1 0 01-.618-1.902l4.755-1.545a1 1 0 011.26.642z"
        fill="#112035"
        fill-rule="evenodd"
      />
      <path
        clip-rule="evenodd"
        d="M17.707 11.146a6 6 0 00-11.413 3.708 5.973 5.973 0 001.42 2.347 1 1 0 01-1.428 1.4 7.974 7.974 0 01-1.894-3.129m13.315-4.326a6.002 6.002 0 01-1.421 6.056 1 1 0 101.43 1.398 8 8 0 10-13.324-3.128"
        fill="#112035"
        fill-rule="evenodd"
      />
      <path
        d="M1 22h22"
        stroke="#112035"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-miterlimit="10"
        stroke-width="2"
      />
      <path
        clip-rule="evenodd"
        d="M0 13C0 6.373 5.373 1 12 1s12 5.373 12 12c0 4.065-2.027 7.65-5.113 9.818a1 1 0 01-.575.182H5.687a1 1 0 01-.575-.182C2.027 20.65 0 17.065 0 13zM12 3C6.477 3 2 7.477 2 13a9.977 9.977 0 004.012 8h11.976A9.975 9.975 0 0022 13c0-5.523-4.477-10-10-10z"
        fill="#112035"
        fill-rule="evenodd"
      />
    </svg>
  </span>
  <div
    class="c2"
    r="auto"
    size="16"
  >
    <span
      class="c3"
      color="apple"
      rotate="0"
      size="16"
    >
      <svg
        fill="none"
        height="100%"
        viewBox="0 0 24 24"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24Z"
          fill="#008933"
          fill-rule="evenodd"
        />
        <path
          d="M18.4808 6.924C18.1605 6.6585 17.6895 6.70275 17.424 7.02L10.3552 15.5025L6.41625 12.8768C6.0705 12.6458 5.6055 12.7395 5.376 13.0845C5.14575 13.4288 5.23875 13.8953 5.58375 14.1248L10.0837 17.1248C10.2112 17.2088 10.356 17.25 10.4992 17.25C10.7152 17.25 10.9282 17.157 11.0767 16.98L18.5768 7.98C18.8415 7.662 18.798 7.18875 18.4808 6.924V6.924Z"
          fill="white"
        />
      </svg>
    </span>
  </div>
</div>
`;
